loleaflet/src/control/Control.UIManager.js

```diff
@@ -71,16 +75,53 @@ L.Control.UIManager = L.Control.extend({
 		this.map.addControl(L.control.infobar());
 		this.map.addControl(L.control.userList());
 
+		var map = this.map;
+
+		var openBusyPopup = function(label) {
+			var json = {
+				id: 'busypopup',
+				jsontype: 'dialog',
+				type: 'modalpopup',
+				children: [
+					{
+						id: 'busycontainer',
+						type: 'container',
+						vertical: 'true',
+						children: [
+							{id: 'busyspinner', type: 'spinner'},
+							{id: 'busylabel', type: 'fixedtext', text: label}
+						]
+					}
+				]
+			};
+			map._socket._onMessage({textMsg: 'jsdialog: ' + JSON.stringify(json)});
+		};
+
+		var closeBusyPopup = function() {
+			var json = {
+				id: 'busypopup',
+				jsontype: 'dialog',
+				action: 'close'
+			};
+			map._socket._onMessage({textMsg: 'jsdialog: ' + JSON.stringify(json)});
+		};
+
 		this.map.on('showbusy', function(e) {
-			if (w2ui['actionbar'])
-				w2utils.lock(w2ui['actionbar'].box, e.label, true);
+			if (window.mode.isMobile()) {
+				if (w2ui['actionbar'])
+					w2utils.lock(w2ui['actionbar'].box, e.label, true);
+			} else {
+				closeBusyPopup();
+				openBusyPopup(e.label);
```

Comment:
I fear this will block the UI too frequently. If I remember correctly, showbusy is used during saving, which might be frequent. Best to check how often we use showbusy and avoid blocking if it's too frequent, instead, we can issue the blockui command from WSD. But if showbusy isn't frequent, this can be helpful.

Reply:
no, I'm not blocking the UI on showbusy.
I'm calling showbusy when blocking the UI :)
But used my new popup in showbusy so all will look unified